tests/utils_tests/test_lorem_ipsum.py

```diff
@@ -1,16 +1,106 @@
+        self.assertEqual(len(words(500).split()), 500)
+
+    @patch('django.utils.lorem_ipsum.random.sample')
+    def test_not_common_words(self, mock_sample):
+        """Words return the values expected."""
+        mock_sample.return_value = ['exercitationem', 'perferendis']
+        self.assertEqual(words(2, False), 'exercitationem perferendis')
+
+    def test_sentence_starts_with_capital(self):
+        """Sentence starts with a capital letter and the second letter is lower case."""
+        self.assertTrue(sentence()[0].isupper())
+        self.assertTrue(sentence()[1].islower())
+
+    @patch('django.utils.lorem_ipsum.random.sample')
+    @patch('django.utils.lorem_ipsum.random.choice')
+    @patch('django.utils.lorem_ipsum.random.randint')
+    def test_sentence(self, mock_randint, mock_choice, mock_sample):
+        """Sentence is built using two phrases and a set of words"""
+        mock_randint.return_value = 2  # split sentence in two; adds comma
+        mock_sample.return_value = ['exercitationem', 'perferendis']
+        mock_choice.return_value = '?'
+
+        sentence_content = sentence()
+        self.assertEqual(mock_randint.call_count, 3)
+        self.assertEqual(mock_sample.call_count, 2)
+        self.assertEqual(mock_choice.call_count, 1)
+        self.assertEqual(sentence_content, 'Exercitationem perferendis, exercitationem perferendis?',)
+
+    @patch('django.utils.lorem_ipsum.random.choice')
+    def test_sentence_ends_well(self, mock_choice):
+        """Sentences end with a question mark or a period"""
+        mock_choice.return_value = '?'
+        self.assertIn(sentence()[-1], '?')
+
+        mock_choice.return_value = '.'
+        self.assertIn(sentence()[-1], '.')
+
+    @patch('django.utils.lorem_ipsum.random.sample')
+    @patch('django.utils.lorem_ipsum.random.choice')
+    @patch('django.utils.lorem_ipsum.random.randint')
+    def test_paragraph(self, mock_paragraph_randint, mock_choice, mock_sample):
+        """Single paragraph generation"""
+
+        # This will override the creation of 2 sentences with 2 phrases
+        mock_paragraph_randint.return_value = 2
+        mock_sample.return_value = ['exercitationem', 'perferendis']
+        mock_choice.return_value = '.'
+
+        par = paragraph()
+        self.assertIsInstance(par, str)
+        expected_paragraph = 'Exercitationem perferendis, exercitationem perferendis. ' \
+                             'Exercitationem perferendis, exercitationem perferendis.'
+        self.assertEqual(mock_paragraph_randint.call_count, 7)
+        self.assertEqual(par, expected_paragraph)
+
+    @patch('django.utils.lorem_ipsum.random.sample')
+    @patch('django.utils.lorem_ipsum.random.choice')
+    @patch('django.utils.lorem_ipsum.random.randint')
+    def test_paragraphs_not_common(self, mock_randint, mock_choice, mock_sample):
+        """Creation of one paragraph not using the common paragraph COMMON_P """
```

Comment:
Docstring shouldn't have a trailing space. Docstring sentences should end with a period.